### PR TITLE
build(deps): update dependencies

### PR DESCRIPTION
**Pull request by lengau**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `make lint`?
- [x] Have you successfully run `pytest tests/unit`?

-----

### CODE REVIEW

**Review by syu-w — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def recursive_data_files(directory, install_directory):
     "pydocstyle",
     "pyftpdlib",
     "pyinstaller; sys_platform == 'win32'",
-    "pylint",
+    "pylint<3",
```
Comment (mr-cal) [inline setup.py:77]: Pylint 3 was released in october and I never noticed!  I assume there are some breaking changes with snapcraft?

File: requirements-devel.txt [hunk 2]
```diff
@@ -128,5 +128,4 @@ zipp==3.17.0
 zope.deprecation==5.0
 zope.interface==6.1
 python-apt @ https://launchpad.net/ubuntu/+archive/primary/+sourcefiles/python-apt/2.4.0ubuntu1/python-apt_2.4.0ubuntu1.tar.xz; sys.platform == "linux"
-setuptools<66
```
Comment (mr-cal) [inline requirements-devel.txt:131]: Question.  I know version 66 as a maximum was fixed with https://github.com/snapcore/snapcraft/pull/4490/files#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7L130, but should we still pin it?  Or is setuptools a special case?

**Review by mr-cal — APPROVED** [review]
Thanks!

File: setup.py (hunk 1 above)
Comment (lengau) [reply setup.py:77]: It's not a huge break, just didn't want to fix linting issues in this PR.

File: requirements-devel.txt (hunk 2 above)
Comment (lengau) [reply requirements-devel.txt:131]: We don't actually need setuptools with a new enough pip. It's still a build dependency, but that happens in a separate environment. I think this was pip 21? (jammy includes 22)

**Pull request merged**